processing/src/main/java/org/apache/druid/frame/field/FieldWriters.java

```diff
@@ -139,7 +143,8 @@ private static FieldWriter makeStringArrayWriter(
       final boolean removeNullBytes
   )
   {
-    final ColumnValueSelector<?> selector = selectorFactory.makeColumnValueSelector(columnName);
+    final ColumnValueSelector<?> selector =
+        TypeCastSelectors.makeColumnValueSelector(selectorFactory, columnName, ColumnType.STRING_ARRAY);
```

Comment:
Won't the code throw with the array writers since the `TypeCastSelectors.makeColumnValueSelector` isn't implemented for any non-numeric type?

Reply:
For arrays, it will fall through to return the underlying selector.

Reply:
I mean, it'll fall through when the desired type is an array.